syntax = "proto3";

package todo;

service Todo {
    rpc createTodo (TodoItem) returns (TodoItem);
    rpc readTodos (TodoRequest) returns (TodoItems);
}

message TodoItem {
    int32 id = 1;
    string text = 2;
}

message TodoItems {
    repeated TodoItem items = 1;
}

message TodoRequest {
    optional int32 id = 1;
}